Fix a bug that makes single point stroke pressureless

bhennion opened the pull request:
This supposedly fixes #1534:

![dots](https://user-images.githubusercontent.com/17818041/130679739-55855fd0-9be4-4ac2-ad96-d89bbf15cfd5.png)

Some workarounds are needed for pressureless devices as well and pressure inference. I will add them later if this works on all available pressure sensitive devices.

Testing on various devices is necessary. For anyone trying this PR out:

- it's easier to get single-point strokes when zoomed in to maximum.
- if you open `Xournal++` from a terminal, you get a "single dot" message for every such single-point stroke.

ViniciusGiroto commented:
Seems to be working. It is kinda hard to draw the thicker dots, but that is probably my inability.

![image](https://user-images.githubusercontent.com/9205380/130707251-7985746d-a810-48cf-a16e-7df016ab17a7.png)
(some of these are not dots)

Tested with a Wacon One on Fedora 34

bhennion commented:
Uh... The seems to be an issue downloading portaudio for Windows, making the build fail
```
--2021-08-24 19:59:47--  (try:20)  http://files.portaudio.com/archives/pa_stable_v190700_20210406.tgz
Connecting to files.portaudio.com (files.portaudio.com)|107.22.210.58|:80... failed: Connection timed out.
Giving up
```
Hopefully it's just a server down. I will rerun it in a day.

personalizedrefrigerator commented:
> So just a fixed dot width for when pressure inference is on then. I still don't know how to find out if the device is using pressure inference from within StrokeHandler though.

It looks like there's a bug in pressure inference!
![image](https://user-images.githubusercontent.com/46334387/131205501-8fdd3e87-6460-4bfd-bebb-bda3291db6a5.png)

For large <i>Δt</i>s (Δt is in tenths of a second), pressure inference estimates the same pressure.

Note that previously, the maximum `dt` considered by pressure inference was `2.0` (0.2 seconds):
```c++ 
...

        double dt = std::min((pos.timestamp - lastPos.timestamp) / 10.0, 2.0);
        double distance = utl::Point<double>(pos.x, pos.y).distance(utl::Point<double>(lastPos.x, lastPos.y));
        double inverseSpeed = (dt) / (distance + 0.1);

...
```

Even with the `std::min` removed, however, it's still giving the same final pressure (on mouse up). As such, the solution here is probably fixing pressure inference for large `Δt`s.

personalizedrefrigerator commented:
Even with a better formula, however, the single-point stroke doesn't seem to change size with respect to the second point's pressure...

bhennion commented:
> Even with a better formula, however, the single-point stroke doesn't seem to change size with respect to the second point's pressure...

That's to be expected with the previous commits. With pressure inference on, you are (for now) supposed to get the same dot as with pressure value 1 every time.
The ButtonRelease event's pressure was never read (on the devices I have access to, this event's pressure value is always 0).

I added a line testing if this value is bigger than the recorded pressure. I also added a commit with your patch.
It seems to works. The bigger question is: how does it feels when writing using pressure inference? Are the dots "the right size"?

LittleHuba reviewed (approved):
Merge in 24 hours into v1.1.1 if there are no objections.
